pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/TableConfigUtils.java

```diff
@@ -744,17 +744,18 @@ static void validateUpsertAndDedupConfig(TableConfig tableConfig, Schema schema)
         Preconditions.checkState(upsertConfig.getMetadataTTL() == 0,
             "enableDeletedKeysCompactionConsistency and metadataTTL shouldn't exist together for upsert table");
 
-        // enableDeletedKeysCompactionConsistency shouldn't exist with enablePreload
-        Preconditions.checkState(!upsertConfig.isEnablePreload(),
-            "enableDeletedKeysCompactionConsistency and enablePreload shouldn't exist together for upsert table");
+        // enableDeletedKeysCompactionConsistency shouldn't exist with preload enabled
+        Preconditions.checkState(upsertConfig.getPreload() != Enablement.ENABLE,
```

Comment:
should we check `== Enablement.DISABLE` here? as iiuc, if it's non-DISABLE, then it'd use server config

Reply:
It cannot be `ENABLE`, but can be `DEFAULT` or `DISABLE`. After server override, we need it to be disabled